torch/testing/_internal/distributed/nn/api/remote_module_test.py

```diff
@@ -208,6 +211,25 @@ def test_remote_parameters(self):
             self.assertEqual(len(param_rrefs), 1)
             self.assertTrue(torch.equal(param_rrefs[0].to_here(), _PARAM_VAL))
 
+    @dist_utils.dist_init
+    def test_device(self):
+        if self.rank != 0:
+            return
+        dst_worker_name = dist_utils.worker_name((self.rank + 1) % self.world_size)
+
+        device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        for remote_module in self._create_remote_module_iter(
+            dst_worker_name, modes=[ModuleCreationMode.MODULE_CTOR], device=device
+        ):
+            for param_rref in remote_module.remote_parameters():
+                param = param_rref.to_here()
+                if torch.cuda.is_available():
+                    self.assertEqual(param.device.type, "cuda")
+                    self.assertEqual(param.device.index, 0)
```

Comment:
nit: Probably better to perform an RPC on the remote node and retrieve the device type from there. This just adds an extra safety net to ensure `to_here` isn't messing up something.

Reply:
Do you want me to add an RPC just in this test file or create a new method remote_device() in RemoteModule?

Reply:
Just add an RPC in this test file to retrieve the remote device type.